pkg/kubefedctl/federate/federate.go

```diff
@@ -154,11 +167,28 @@ func (j *federateResource) Run(cmdOut io.Writer, config util.FedConfig) error {
 		return errors.Wrap(err, "Failed to get host cluster config")
 	}
 
+	if len(j.filename) > 0 {
+		resources, err := decodeUnstructuredFromFile(j.filename)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to load yamls from file %q", j.filename)
+		}
+		federatedResources, err := federateResources(resources)
+		if err != nil {
+			return err
```

Comment:
any rule to use `errrors.Wrapf` or `return err`? It seems not consistent in code context.